### PR TITLE
Fix dependendency issues with commons logging

### PR DESCRIPTION
**Pull request by Siedlerchr**
Update gemfx
update unirest to 4.x  and update http client 

Fixes https://github.com/JabRef/jabref/issues/11401
Fixes https://github.com/JabRef/jabref/issues/11397
Fixes https://github.com/JabRef/jabref/issues/11398

PubMed import works now: tested with file: 
[pubmed-36096847.txt](https://github.com/user-attachments/files/15930162/pubmed-36096847.txt)

### Mandatory checks

- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by ryan-carpenter** [comment]
PubMed import also works for me after installing today's rpm.

File: build.gradle [hunk 1]
```diff
@@ -145,10 +145,10 @@ dependencies {
 
     def pdfbox = "3.0.2"
     implementation ("org.apache.pdfbox:pdfbox:$pdfbox") {
-        exclude group: 'commons-logging'
+       exclude group: 'commons-logging'
     }
     implementation ("org.apache.pdfbox:fontbox:$pdfbox") {
-        exclude group: 'commons-logging'
+         exclude group: 'commons-logging'
```
Comment (koppor) [inline build.gradle:151]: Please fix spacing change

File: build.gradle [hunk 2]
```diff
@@ -145,10 +145,10 @@ dependencies {
 
     def pdfbox = "3.0.2"
     implementation ("org.apache.pdfbox:pdfbox:$pdfbox") {
-        exclude group: 'commons-logging'
+       exclude group: 'commons-logging'
```
Comment (koppor) [inline build.gradle:148]: Please fix spacing change

File: build.gradle [hunk 3]
```diff
@@ -222,9 +223,11 @@ dependencies {
     }
     implementation 'org.fxmisc.flowless:flowless:0.7.3'
     implementation 'org.fxmisc.richtext:richtextfx:0.11.2'
-    implementation (group: 'com.dlsc.gemsfx', name: 'gemsfx', version: '2.25.0') {
+    implementation (group: 'com.dlsc.gemsfx', name: 'gemsfx', version: '2.32.0') {
         exclude module: 'javax.inject' // Split package, use only jakarta.inject
         exclude module: 'commons-lang3'
+        exclude group: 'org.apache.commons.validator' // creates issues with common-logging
```
Comment (koppor) [inline build.gradle:229]: No need for the comment. If kept, place in next line

File: build.gradle [hunk 4]
```diff
@@ -235,9 +238,11 @@ dependencies {
     implementation 'org.controlsfx:controlsfx:11.2.1'
 
     implementation 'org.jsoup:jsoup:1.17.2'
-    implementation 'com.konghq:unirest-java:3.14.5'
+    implementation 'com.konghq:unirest-java-core:4.4.0'
+    implementation 'com.konghq:unirest-modules-gson:4.4.0'
+    implementation 'org.apache.httpcomponents.client5:httpclient5:5.3.1'
 
-    implementation 'org.slf4j:slf4j-api:2.0.13'
+    implementation "org.slf4j:slf4j-api:2.0.13"
```
Comment (koppor) [inline build.gradle:245]: I think, all dependencies should be in ' not in "

Double check with dependabot. They IMHO can do only '

File: build.gradle [hunk 5]
```diff
@@ -712,12 +717,14 @@ jlink {
         requires 'jakarta.xml.bind'
         requires 'org.apache.commons.lang3'
         requires 'org.apache.commons.text'
+        requires 'org.apache.commons.logging';
```
Comment (koppor) [inline build.gradle:720]: Move up one line. Alphabetical ordering - the current order is not logical

**Review by koppor — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 5 above)
Comment (Siedlerchr) [reply build.gradle:720]: The order can be crucial just simple alphabetical  order will not work properly (there is a comment stating this)

File: build.gradle (hunk 4 above)
Comment (Siedlerchr) [reply build.gradle]: Did not find anything int he docs, changed it back to single quote to be consistent

**Comment by calixtus** [comment]
Looks good to me so far.
The gvk fetcher and the BVB fetcher break with a strange error. Everything seems to be in order. I'll look into it later.
Doesn't have to stop merging this pr.

**Review by calixtus — APPROVED** [review]

File: build.gradle [hunk 6]
```diff
@@ -672,7 +677,7 @@ jlink {
     // but it was removed in the newer releases.
     // The pom.xml associated with such a non-modular artifact does not mention that the artifact depends on the removed code
     // (because the artifact was published when this code was still available in the JDK).
-    forceMerge "controlsfx", "bcprov", "jaxb", "istack", "stax", "log4j"
```
Comment (koppor) [inline build.gradle:675]: We have to try if logging works for a library logging using log4j. I don't know how to find out which one is it 😅🙈

**Review by koppor — APPROVED** [review]
Lets merge and fix later if an issue occurs 😅

File: build.gradle (hunk 6 above)
Comment (Siedlerchr) [reply build.gradle:675]: I played around in Jabref and tested different dialogs and actions and so far I did not stumble across any error

**Pull request merged**